### PR TITLE
Everywhere: Fix discarded null checks [nightly Clang fix]

### PR DESCRIPTION
**Pull request by BertalanD**
The compiler is able to statically prove that these comparisons always return a certain value or have attributes that make it assume so, so with optimizations enabled (which we do), these code paths are discarded. Don't lie to ourselves by not removing them.

### CODE REVIEW

File: Userland/Libraries/LibC/stdio.cpp [hunk 1]
```diff
@@ -25,6 +25,10 @@
 #include <syscall.h>
 #include <unistd.h>
 
+// Used for getting around the implicit `nonnull` parameter attributes added by GCC.
+// These made explicit null pointer checks within the function into no-ops.
+#define alias(old, new) extern __typeof__(old) new __attribute__((alias(#old)))
```
Comment (alimpfard) [inline Userland/Libraries/LibC/stdio.cpp:30]: This is c++ code, can you use `decltype()` instead?

File: Userland/Libraries/LibC/stdio.cpp [hunk 2]
```diff
@@ -788,6 +792,8 @@ int fputc(int ch, FILE* stream)
     return byte;
 }
 
+alias(__fputc, fputc);
```
Comment (alimpfard) [inline Userland/Libraries/LibC/stdio.cpp:795]: Doing this to only some of the functions is weird, and doing it to all of them sounds like a waste...
Is there no other way to disable the nonnull attribute here? e.g. by manually giving it some attributes, maybe?

Comment (BertalanD) [reply Userland/Libraries/LibC/stdio.cpp:795]: The documentation seems to indicate that it's impossible if nonnull is applied, which GCC does automatically.

My only other idea is to compile this specific file with `-fno-builtin=fread -fno-builtiin=fputc -fno-builtin=fputs`

Comment (BertalanD) [reply Userland/Libraries/LibC/stdio.cpp]: IIRC there is no such pragma, only `diagnostic`, `target`, `optimize` and `unroll` can be configured via the preprocessor.

**Review by dascandy — APPROVED** [review]
LGTM

**Pull request merged**